### PR TITLE
Support Java 18

### PR DESCRIPTION
**Pull request by kaklakariada**

### CODE REVIEW

File: api/src/main/java/org/itsallcode/openfasttrace/api/importer/SpecificationListBuilder.java [hunk 1]
```diff
@@ -146,6 +143,7 @@ public List<SpecificationItem> build()
     }
 
     /**
+     * Get the total number of items.
      * @return the total number of items.
```
Comment (redcatbear) [inline api/src/main/java/org/itsallcode/openfasttrace/api/importer/SpecificationListBuilder.java:147]:
```suggestion
     * Get the total number of items.
     *
     * @return the total number of items.
```

File: api/src/main/java/org/itsallcode/openfasttrace/api/importer/tag/config/DescribedPathMatcher.java [hunk 2]
```diff
@@ -62,6 +60,7 @@ private static String addMissingPatternPrefix(final String pattern)
     }
 
     /**
+     * Get the description for this {@link DescribedPathMatcher}.
      * @return the description for this {@link DescribedPathMatcher}.
```
Comment (redcatbear) [inline api/src/main/java/org/itsallcode/openfasttrace/api/importer/tag/config/DescribedPathMatcher.java:64]:
```suggestion
     * Get the description for this {@link DescribedPathMatcher}.
     *
     * @return the description for this {@link DescribedPathMatcher}.
```

File: importer/tag/src/main/java/org/itsallcode/openfasttrace/importer/tag/TagImporterFactory.java [hunk 3]
```diff
@@ -29,7 +26,8 @@ public class TagImporterFactory extends ImporterFactory
             "json", "htm", "html", "xhtml", "yaml", // markup languages
             "java", // Java
             "clj", "kt", "scala", // JVM languages
-            "js", // Java script
+            "js", // JavaScript
+            "ts", // TypeScript
```
Comment (redcatbear) [inline importer/tag/src/main/java/org/itsallcode/openfasttrace/importer/tag/TagImporterFactory.java:30]: Secretly sneaked Typescript in there? :-)

Did you add that to the handbook too?

**Review by redcatbear — CHANGES_REQUESTED** [review]
Please don't forget to list TS in the handbook.

File: importer/tag/src/main/java/org/itsallcode/openfasttrace/importer/tag/TagImporterFactory.java (hunk 3 above)
Comment (kaklakariada) [reply importer/tag/src/main/java/org/itsallcode/openfasttrace/importer/tag/TagImporterFactory.java:30]: ✅

File: doc/CHANGELOG.md [hunk 4]
```diff
@@ -24,6 +24,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fixed static code analysis warnings in unit tests [PR #329](https://github.com/itsallcode/openfasttrace/pull/329)
 - Upgrade test and build dependencies [PR #330](https://github.com/itsallcode/openfasttrace/pull/330)
 - Removed license headers from sources and license-maven-plugin [PR #331](https://github.com/itsallcode/openfasttrace/pull/331)
+- Added support for Java 18 [PR #344](https://github.com/itsallcode/openfasttrace/pull/344)
+  - Added support for tracing TypeScript files (`.ts`)
```
Comment (redcatbear) [inline doc/CHANGELOG.md:28]: :)

File: parent/pom.xml [hunk 5]
```diff
@@ -244,6 +246,17 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>java18</id>
+            <activation>
+                <jdk>18</jdk>
```
Comment (redcatbear) [inline parent/pom.xml:252]: :+1:

**Review by redcatbear — APPROVED** [review]

**Pull request merged**